internal/ext/exporter.go

```diff
@@ -53,198 +59,220 @@ func versionString(v semver.Version) string {
 func (e *Exporter) Export(ctx context.Context, w io.Writer) error {
 	var (
 		enc       = yaml.NewEncoder(w)
-		doc       = new(Document)
 		batchSize = e.batchSize
 	)
 
-	doc.Version = versionString(latestVersion)
-	doc.Namespace = e.namespace
-
-	defer enc.Close()
-
-	var (
-		remaining = true
-		nextPage  string
-	)
+	var namespaces = e.namespaces
 
-	// export flags/variants in batches
-	for batch := int32(0); remaining; batch++ {
-		resp, err := e.store.ListFlags(
-			ctx,
-			&flipt.ListFlagRequest{
-				NamespaceKey: e.namespace,
-				PageToken:    nextPage,
-				Limit:        batchSize,
-			},
-		)
+	// If allNamespaces is "true", then retrieve all the namespaces, and store them in a string slice.
+	if e.allNamespaces {
+		ns, err := e.store.ListNamespaces(ctx, &flipt.ListNamespaceRequest{})
```

Comment:
we should probably handle pagination here if they have a ton of namespaces

Reply:
@markphelps Ahh indeed. Will do that.